LinkList.java

```diff
@@ -120,34 +133,153 @@ public boolean contains(Object o) {
 
      }
 
-     public static void main(String [ ] args) {
-        ArrayList a=new LinkList();
+      public Iterator<E> iterator() {
+        return null;
+    }  
+
+    public void addFirst(E e) {
+        if(size()==0){
+            head.data=e;
+            numNodes++;
+            return;
+        }
```

Comment:
Would it be easier if you create a newHead and then newHead.next = head; head = newHead;

Reply:
good tip